client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -37,6 +38,13 @@ const PatternAssembler: Step = ( { navigation } ) => {
 	const siteId = useSiteIdParam();
 	const siteSlugOrId = siteSlug ? siteSlug : siteId;
 
+	useEffect( () => {
+		// Require to start the flow from the first step
+		if ( ! selectedDesign ) {
+			location.href = `/setup${ location.search }`;
```

Comment:
Suggestion: Use [goToStep()](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/site-setup-flow.ts#L577) from Stepper's `navigation` instead of `location.href` to avoid redoing HTTP requests. What do you think?